tendermint/src/lite_impl/signed_header.rs

```diff
@@ -82,7 +82,11 @@ fn non_absent_votes(commit: &block::Commit) -> Vec<vote::Vote> {
                     vote_type: vote::Type::Precommit,
                     height: commit.height,
                     round: commit.round,
-                    block_id: Option::from(commit.block_id.clone()),
+                    block_id: if commit_sig.block_id_flag == BlockIDFlag::BlockIDFlagCommit {
+                        Some(commit.block_id.clone())
+                    } else {
+                        None
+                    },
```

Comment:
Slightly more idiomatic would be:
```suggestion
                    block_id: match commit_sig.block_id_flag {
                        BlockIDFlag::BlockIDFlagCommit => Option::from(commit.block_id.clone()),
                        _ => None,
                    },
```